### PR TITLE
Allow a server to be provided, e.g. so you can configure an HTTPS one.

### PR DESCRIPTION
**Pull request by insightfuls**
Allow a server to be provided, e.g. so you can configure an HTTPS one.

The Koa shortcut method uses plain HTTP.

Cf. https://koajs.com/#example-5

### CODE REVIEW

**Comment by kierans** [comment]
Can you please update the description with a use case for this change please. What does it improve upon?

File: src/simple-web.js [hunk 1]
```diff
@@ -30,37 +30,44 @@ const PASSTHROUGH_CONFIG_KEYS = [
...
 		});
 	}
 
 	stop() {
 		return new Promise((resolve, reject) => {
-			if (this._server) {
+			if (this._server && this._server.listening) {
 				this._server.on("close", () => {
-					this._server = null;
+					this._server.off("error", reject);
```
Comment (kierans) [inline src/simple-web.js:70]: Where do you ever start listening for the `error` event with the `reject` callback? I don't think this is necessary

Comment (insightfuls) [reply src/simple-web.js:70]: It's in the existing code at line 68 (now line 75).

Comment (kierans) [reply src/simple-web.js:70]: That will teach me for trying to read the changes on my phone.

**Comment by kierans** [comment]
@insightfuls If one is providing a `server`, shouldn't it be the responsibility of the caller to set the relevant `config` keys?

**Comment by kierans** [comment]
@insightfuls Given the responsibilities are split between the caller and `koa-simple-web` I'm going to push the configuration of the server back onto the caller. This will be a breaking change, so this PR will be part of version `2.0.0`.

**Pull request merged**